harness/tests/determined/common/experimental/test_determined.py

```diff
@@ -34,7 +36,7 @@ def _make_client() -> Determined:
 
 
 @responses.activate
-def test_default_retry(make_client: Callable[[], Determined]) -> None:
+def test_default_retry_max_retries(make_client: Callable[[], Determined]) -> None:
```

Comment:
i split this test into 2:
`test_default_retry_retries_until_max` and `test_default_retry_fails_after_max_retries`. i kept the assertion that `The default retry logic will fail at its 6th call.` in the second test, because it seems like if it's a separate test, it should verify that it called the REST API 5 times before giving up. i suppose the assert isn't strictly necessary since we have the first `test_default_retry_retries_until_max` test though. if you have a strong preference to remove it, happy to do so.

Reply:
I'm ok with 3 being tested, but I still think that the second test is testing two unrelated behaviors (2 and 3 from before).